### DISCUSSION TITLE
Inputs

### DISCUSSION THREAD

**Discussion by Yiquan-lol** [praveen-palanisamy/macad-agents#16] [Q&A]
hello
Is the input size [168,168,3] or [84,84,3] in [https://github.com/praveen-palanisamy/macad-agents/blob/master/src/macad_agents/rllib/impala_multiagent_shared_weights.py](url)？
In addition, conv_filter is [[32, [8], 4], [64, [4], 2], [64, [3, 3], 1)]?

**Comment by praveen-palanisamy** [comment]
The observation for the `HomoNcomIndePOIntrxMASS3CTWN3` [MACAD-Gym](https://github.com/praveen-palanisamy/macad-gym) environment as per the environment config is `[168, 168, 3]`. This is the shape of the observation sent to the [Impala_multiagent_shared_weights.py](https://github.com/praveen-palanisamy/macad-agents/blob/master/src/macad_agents/rllib/impala_multiagent_shared_weights.py) agent trainer script. This script resizes the observations to `[84, 84, 3]` before feeding the agent. 
The Impala agent's CNN policy IIRC, uses the following conv filter configuration:

```python
[[16, [8], 4],
[32, [4], 2],
[256, [11], 1]]
```

You can customize the shape of the Agent's input by changing these lines:
https://github.com/praveen-palanisamy/macad-agents/blob/b2726b391ee4f6844e2ae80883fb99f8027d34b4/src/macad_agents/rllib/impala_multiagent_shared_weights.py#L71-L82

You can also customize the filters used by un-commenting and modifying these lines:

https://github.com/praveen-palanisamy/macad-agents/blob/b2726b391ee4f6844e2ae80883fb99f8027d34b4/src/macad_agents/rllib/impala_multiagent_shared_weights.py#L107-L109
